src/component-library/TokenField/TokenField.tsx

```diff
@@ -1,27 +1,59 @@
-import * as React from 'react';
+import { forwardRef, useEffect, useState } from 'react';
 
 import { NumberInputProps } from '../NumberInput';
 import { Stack } from '../Stack';
-import { TokenBalance } from '../TokenBalance';
+import { useDOMRef } from '../utils/dom';
+import { triggerChangeEvent } from '../utils/input';
 import {
+  StyledTokenBalance,
   TokenAdornment,
   TokenFieldInnerWrapper,
   TokenFieldInput,
   TokenFieldSymbol,
   TokenFieldUSD
 } from './TokenField.style';
 
-interface TokenFieldProps extends NumberInputProps {
-  balance?: {
-    value: string;
-    valueInUSD: string;
-  };
+type Props = {
   tokenSymbol: string;
   valueInUSD: string;
-}
+  balance?: string | number;
+  balanceInUSD?: string | number;
+};
+
+type InheritAttrs = Omit<NumberInputProps, keyof Props>;
+
+type TokenFieldProps = Props & InheritAttrs;
+
+const TokenField = forwardRef<HTMLInputElement, TokenFieldProps>(
+  (
+    {
+      tokenSymbol,
+      valueInUSD,
+      balance,
+      balanceInUSD,
+      value: valueProp,
+      isDisabled,
+      className,
+      style,
+      hidden,
+      ...props
+    },
+    ref
+  ): JSX.Element => {
+    const inputRef = useDOMRef(ref);
+    const [value, setValue] = useState<number | undefined>(valueProp);
+
+    const handleClickBalance = () => {
+      const newValue = Number(balance);
+      setValue(newValue);
+      triggerChangeEvent(inputRef, newValue);
+    };
```

Comment:
I'm not sure I completely follow the code logic here. Why is there need to manually trigger the change event and control the value of the input element with React state?